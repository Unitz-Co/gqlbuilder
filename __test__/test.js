const _ = require('lodash');
const { GqlBuilder } = require('../index');

describe('gqlbuilder', () => {
  const query = GqlBuilder.from('QueryNewUsers');

  test('update alias', () => {
    console.snapshot(query.clone().update('user.presence', { alias: 'newalPre' }).toString());

    console.snapshot(query.clone().update('user', { alias: 'myuser' }).toString());

    console.snapshot(query.clone().update({ alias: 'rootuser' }).toString());
  });

  test('set selection', () => {
    console.snapshot(query.clone().update({ selections: 'user { id }' }).toString());

    console.snapshot(query.clone().update({ selections: 'user { id profile { id display_name } }' }).toString());

    console.snapshot(
      query
        .clone()
        .update({ selections: () => 'user { id profile { id display_name } }' })
        .toString(),
    );

    console.snapshot(query.clone().update('user', { selections: 'user_by_pk { id }' }).toString());

    console.snapshot(
      query.clone().update('user', { selections: 'user_by_pk { id profile { id display_name } }' }).toString(),
    );

    console.snapshot(query.clone().update('user', { selections: 'user_by_pk(id: "user_id") { id }' }).toString());

    console.snapshot(
      query
        .clone()
        .update('user', { selections: 'user_by_pk(id: "user_id") { id profile { id display_name } }' })
        .toString(),
    );

    console.snapshot(
      query
        .clone()
        .update('user', { selections: () => 'user_by_pk(id: "user_id") { id profile { id display_name } }' })
        .toString(),
    );

    console.snapshot(query.clone().update('user.profile', { selections: 'id display_name avatar_url' }).toString());

    console.snapshot(
      query.clone().update('user.profile', { selections: 'id display_name avatar_url(limit: 1)' }).toString(),
    );

    console.snapshot(
      query
        .clone()
        .update('user.profile', { selections: () => 'id display_name avatar_url(limit: 1)' })
        .toString(),
    );
  });

  // test('update alias nested selection', () => {
  //   // expect(sum(1, 2)).toMatchSnapshot();
  //   const myQuery = query.clone()
  //     .update('user.presence', { alias: 'newalPre' })
  //   console.snapshot(myQuery.toString());

  //   const myQuery = query.clone()
  //     .update('user', { alias: 'myuser' })

  //     console.snapshot(myQuery.toString());

  // });
});

// const myQuery = query.clone()
//   .update('user.presence', { alias: 'newalPre' })
//   .update('user', { alias: 'newUser' })
//   .update('user', { arguments: ({ node }) => node.merge(`where: {uid: "123123", presence: {status: {_eqt: "newxasd", _ne: "masdbcsdc", _if: { _t: "123213" }}}}`) })
//   .update('user', { arguments: ({ node }) => node.remove(`where.presence.status._eqt`) })
//   // .update('user', { arguments: ({ node }) => node.remove(`where`) })
//   // .update('user', { arguments: ({ node }) => node.set(`where`) })
//   // .update('user', { arguments: ({ node }) => node.set('where.presence', `presencexy: { status: "12332" }`)
//   .update('user', { arguments: ({ node }) => node.merge(({ presencexxx: { x: 1, y: 234 }})) })
//   // .update('user', { arguments: ({ node }) => node.set('where.presence', ({
//   //   presence: {
//   //     status: "12341234",
//   //     boolTrue: true,
//   //     boolFalse: false,
//   //     number0: 0,
//   //     numberAny: 1234,
//   //     stringQuote: 'quote',
//   //     stringDoubleQuote: "quoqoq",
//   //     numberText: '1',
//   //     numberText0: '0',
//   //     arr: [1, 2, "b", "d", { cstr: "cst", cnum: 1 }],
//   //     nestedObj: {
//   //       nestProp: '1',
//   //     }
//   //   },
//   //   sibling: "number",
//   // })) })

//   .update('user.presence', { arguments: `order: "ast"` })
//   .update('user.presence', { arguments: ({ node }) => node.merge(`sort: "xyz"`) })
//   .update('user.presence', 'arguments', ({ node }) => node.merge(`sort: "xyzt"`))
//   // .update('user.presence', { selections: ({ node }) => node.merge(`id, name, status`) })
//   // .update('user.presence', { selections: ({ node }) => node.merge(`id, name`) })
//   .update('user.presence', 'selections', ({ node }) => node.merge(`id, name`) )
//   .update('user.presence', { selections: ({ node }) => node.merge(`name`) })
//   // .update('user.presence', { selections: ({ node }) => node.merge(`id, status`) })
//   .update('user.presence', { selections: ({ node }) => node.merge(`status { lastUpdate }`) })
//   .update('user.presence', { selections: ({ node }) => node.merge(`status { isOffline }`) })
//   .update('user.presence', { selections: ({ node }) => node.merge(`node1 {id, name, status}`) })
//   .update('user.presence', { selections: ({ node }) => node.merge(`node2 {id, name}`) })
//   .update('user.presence', { selections: ({ node }) => node.merge(`node2 {status}`) })
//   // .update('user', { selections: ({ node }) => node.remove(`presence.name`) })
//   .update('user', { selections: ({ node }) => node.remove(`presence { id }`) })
//   .update('user', { selections: ({ node }) => node.remove(`presence { node2 { id, name } }`) })
//   .update('user.presence', { selections: ({ node }) => node.set('node1', `node1 { new }`) })
//   // .update('user', { selections: ({ node }) => node.add(`newobjec.id`) })

// console.log('myQuerymyQuery', myQuery.toString());
