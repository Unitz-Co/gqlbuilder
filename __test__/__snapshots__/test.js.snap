// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gqlbuilder set selection 1`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    user {
      id
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 2`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    user {
      id
      profile {
        id
        display_name
      }
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 3`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    user {
      id
      profile {
        id
        display_name
      }
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 4`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    user_by_pk {
      id
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 5`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    user_by_pk {
      id
      profile {
        id
        display_name
      }
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 6`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    user_by_pk(id: \\"user_id\\") {
      id
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 7`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    user_by_pk(id: \\"user_id\\") {
      id
      profile {
        id
        display_name
      }
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 8`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    user_by_pk(id: \\"user_id\\") {
      id
      profile {
        id
        display_name
      }
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 9`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    id
    presence {
      id
      status
    }
    profile {
      id
      display_name
      avatar_url
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 10`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    id
    presence {
      id
      status
    }
    profile {
      id
      display_name
      avatar_url(limit: 1)
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder set selection 11`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    id
    presence {
      id
      status
    }
    profile {
      id
      display_name
      avatar_url(limit: 1)
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder update alias 1`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    id
    newalPre: presence {
      id
      status
    }
    profile {
      id
      display_name
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder update alias 2`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  myuser: user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    id
    presence {
      id
      status
    }
    profile {
      id
      display_name
    }
  }
}
",
  ],
}
`;

exports[`gqlbuilder update alias 3`] = `
Object {
  "args": Array [
    "query QueryNewUsers {
  rootuser: user(where: {presence: {status: {_eq: \\"new\\"}}}) {
    id
    presence {
      id
      status
    }
    profile {
      id
      display_name
    }
  }
}
",
  ],
}
`;
